app/client/src/pages/Editor/PropertyPane/PropertyControlsGenerator.tsx

```diff
@@ -29,11 +31,26 @@ export interface PropertyControlsGeneratorProps {
   searchQuery?: string;
 }
 
+export const shouldSectionBeExpanded = (
+  sectionConfig: PropertyPaneSectionConfig,
+  isFlagEnabled: boolean,
+) => {
+  if (isFlagEnabled && "expandedByDefault" in sectionConfig) {
+    return !!sectionConfig.expandedByDefault;
+  } else {
+    if (!!sectionConfig.isDefaultOpen) {
+      return sectionConfig.isDefaultOpen;
+    }
+    return true;
+  }
```

Comment:
Please correct me if I am wrong here, for the condition when isDefaultOpen is defined and if `false`. We would want `shouldSectionBeExpanded` to return false instead it would return true according to the above logic as `!!sectionConfig.isDefaultOpen` will be false and it would not go inside the if condition and instead  `return true` will be executed. Maybe we could write the above logic as 
```suggestion
  if (isFlagEnabled && "expandedByDefault" in sectionConfig) return !!sectionConfig.expandedByDefault;
  
  if ("isDefaultOpen" in sectionConfig ) return sectionConfig.isDefaultOpen;
  
  return true;
  
```

Reply:
Agreed, will update this, Nice catch!